relay-general/src/store/transactions/processor.rs

```diff
@@ -396,6 +414,53 @@ impl Processor for TransactionsProcessor<'_> {
         Ok(())
     }
 
+    /// Validates the length and characters of measurement names.
+    fn process_measurements(
+        &mut self,
+        value: &mut crate::protocol::Measurements,
+        meta: &mut Meta,
+        _state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        let measurements = &mut value.0;
+        let mut removed_measurements = Object::new();
+
+        measurements.retain(|key, value| {
+            let measurement = match value.value_mut() {
+                Some(m) => m,
+                None => return false,
+            };
+
+            if !is_valid_metric_name(key) {
+                removed_measurements
+                    .insert(key.clone(), Annotated::new(std::mem::take(measurement)));
+                return false;
+            }
+
+            let name_len = key.len();
+            let unit_len = measurement.unit.value().map(|val| val.to_string().len());
+
+            if let (Some(fixed_len), Some(max_len), Some(unit_len)) = (
+                self.measurement_config.fixed_length,
+                self.measurement_config.max_len,
+                unit_len,
+            ) {
+                if (name_len + unit_len + fixed_len) > max_len {
+                    removed_measurements
+                        .insert(key.clone(), Annotated::new(std::mem::take(measurement)));
+                    return false;
+                }
+            }
+            true
+        });
+
+        if !removed_measurements.is_empty() {
+            meta.add_error(Error::invalid("Invalid measurement names"));
```

Comment:
nit: It would be nice to get a more specific error message when the name is too long, e.g.

```rust
format!("Measurement name too long: name + unit > {}", limit)
```

Reply:
will do, when figured out how to deal with errors (as described down the page)